examples/2d/elixir_advection_p4est_non_conforming.jl

```diff
@@ -0,0 +1,83 @@
+
+using OrdinaryDiffEq
+using P4est
+using Trixi
+
+
+# Refine bottom left quadrant of each forest to level 4
+function refine_fn(p4est, which_tree, quadrant)
+  if quadrant.x == 0 && quadrant.y == 0 && quadrant.level < 4
+    # return true (refine)
+    return Cint(1)
+  else
+    # return false (don't refine)
+    return Cint(0)
+  end
+end
```

Comment:
Why define it up here if only used in one place below? I'd rather move it down to line 42, right before it is actually used. Also, instead of `_fn` I'd prefer something more descriptive such as `refinement_indicator` etc.

Reply:
I tried to make clear that this is a function that will only be called by p4est. I appended the name for these functions in the p4est docs to the Julia function names. So, functions used as `iter_volume` (named like this in p4est) are named, e.g., `copy_to_quad_iter_volume`.

Reply:
OK, fair enough